src/goto-symex/symex_decl.cpp

```diff
@@ -25,14 +25,11 @@ void goto_symext::symex_decl(statet &state)
 
   const codet &code=to_code(instruction.code);
 
-  if(code.operands().size()==2)
-    throw "two-operand decl not supported here";
-
-  if(code.operands().size()!=1)
-    throw "decl expects one operand";
-
-  if(code.op0().id()!=ID_symbol)
-    throw "decl expects symbol as first operand";
+  // two-operand decl not supported here
+  PRECONDITION(code.operands().size()!=2);
+  // we handle the decl with only one operand
+  PRECONDITION(code.operands().size()==1);
```

Comment:
These preconditions are redundant, I'd suggest removing one of them (the != one).